ctlrender does not compile when OpenEXR is not found

While ILMBASE is a required library for CTL and ctlrender, OpenEXR should be an optional library for CTL and ctlrender.  Currently, ctlrender does not compile when OpenEXR is not found, and gives this compile error message:
```

/usr/bin/ld: CMakeFiles/ctlrender.dir/transform.cc.o: in function `transform(char const*, char const*, float, float, format_t*, 
Compression*, std::__cxx11::list<ctl_operation_t, std::allocator<ctl_operation_t> > const&, std::__cxx11::list<ctl_parameter_t, 
std::allocator<ctl_parameter_t> > const&)':
transform.cc:(.text+0x3d37): undefined reference to `exr_write(char const*, float, ctl::dpx::fb<float> const&, format_t*, Compression*)'
collect2: error: ld returned 1 exit status
make[2]: *** [ctlrender/CMakeFiles/ctlrender.dir/build.make:219: ctlrender/ctlrender] Error 1
make[1]: *** [CMakeFiles/Makefile2:410: ctlrender/CMakeFiles/ctlrender.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/.github/workflows/cmake_no_openexr.yml
+++ b/.github/workflows/cmake_no_openexr.yml
@@ -1,0 +1,64 @@
+name: CMake - No OpenEXR
+
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'doc/**'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'doc/**'
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: remove openexr
+      run: sudo apt-get --purge remove libopenexr-dev -y
+
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+    - name: Build
+      # Build your program with the given configuration
+      run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+  test:
+    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: remove openexr
+      run: sudo apt-get --purge remove libopenexr-dev -y
+
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+    - name: Build
+      # Build your program with the given configuration
+      run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+    - name: Test
+      working-directory: ${{github.workspace}}/build
+      run: make check

--- a/ctlrender/exr_file.cc
+++ b/ctlrender/exr_file.cc
@@ -258,12 +258,17 @@ void exr_write(const char *name, float scale, const ctl::dpx::fb<float> &pixels,
 bool exr_read(const char *name, float scale,
               ctl::dpx::fb<float> *pixels,
               format_t *bpp) {
+//    fprintf(stderr, "WARNING on line %d of file %s in function %s(): exr_read() function was called but OpenEXR library was not found\n",
+//    __LINE__, __FILE__, __FUNCTION__);
 	return FALSE;
 }
 
 void exr_write(const char *name, float scale,
                const ctl::dpx::fb<float> &pixels,
-               format_t *format) {
+               format_t *format,
+               Compression *compression) {
+//    fprintf(stderr, "WARNING on line %d of file %s in function %s(): exr_write() function was called but OpenEXR library was not found\n",
+//    __LINE__, __FILE__, __FUNCTION__);
 }
 
 #endif

--- a/unittest/IlmCtlMath/CMakeLists.txt
+++ b/unittest/IlmCtlMath/CMakeLists.txt
@@ -5,9 +5,6 @@ add_executable(IlmCtlMathTest
     testGaussRec.cpp
 )
 
-include_directories( ${OpenEXR_INCLUDE_DIRS} )
-link_directories( ${OpenEXR_LIBRARY_DIRS} )
-
 include_directories( "${CMAKE_CURRENT_SOURCE_DIR}" 
                      "${PROJECT_SOURCE_DIR}/lib/IlmCtlMath" )
                      

--- a/unittest/IlmImfCtl/CMakeLists.txt
+++ b/unittest/IlmImfCtl/CMakeLists.txt
@@ -1,3 +1,7 @@
+if ( NOT OpenEXR_FOUND )
+    message(WARNING "OpenEXR not found, skipping IlmImfCtl unit test")
+    return()
+endif()
 
 include_directories( ${OpenEXR_INCLUDE_DIRS} )
 link_directories( ${OpenEXR_LIBRARY_DIRS} )

